### DISCUSSION TITLE
v7.0.3

### DISCUSSION THREAD

**Discussion by fcakyon** [fcakyon/yolov5-pip#182] [Announcements]
## 🤗 HuggingFace Hub Integration

- Use yolov5 models from hub:

```python
import yolov5

# load model
model = yolov5.load('fcakyon/yolov5s-v7.0')

# set image
img = 'https://github.com/ultralytics/yolov5/raw/master/data/images/zidane.jpg'

# perform inference
results = model(img, size=640)

# show detection bounding boxes on image
results.show()
```

- Fine-tune yolov5 models from hub:

```bash
yolov5 train --img 640 --batch 16 --weights fcakyon/yolov5s-v7.0 --epochs 10 --device cuda:0
```

- Automatically push fine-tuned weight and training logs to hub (with autogenerated model card):

```bash
yolov5 train --data data.yaml --weights yolov5s.pt --hf_model_id username/modelname --hf_token YOUR-HF-WRITE-TOKEN
```

Available models: https://huggingface.co/models?other=yolov5

## What's Changed
* ad hf hub tests to package testing by @fcakyon in https://github.com/fcakyon/yolov5-pip/pull/177
* fix private hub model download by @fcakyon in https://github.com/fcakyon/yolov5-pip/pull/179
* add automatic hf hub upload, refactor helpers by @fcakyon in https://github.com/fcakyon/yolov5-pip/pull/180
* fix readme by @fcakyon in https://github.com/fcakyon/yolov5-pip/pull/181

**Full Changelog**: https://github.com/fcakyon/yolov5-pip/compare/7.0.2...7.0.3

<hr /><em>This discussion was created from the release <a href='https://github.com/fcakyon/yolov5-pip/releases/tag/7.0.3'>v7.0.3</a>.</em>

**Comment by kadirnar** [comment]
Geliştirmeler için teşekkür ederim. 

**Düzeltme İsteği:** Bu pathi düzeltebilir misin? https://github.com/fcakyon/yolov5-pip/blob/main/yolov5/helpers.py#L333

**Soru:** Helpers.py dosyasındaki train_utils fonksiyonları HF platformunda train yapabilmek için mi?

**Özellik İsteği:** https://huggingface.co/spaces/fcakyon/sahi-yolov5/  reposuna bu desteklere ekleyebilir misiniz? HF platformundaki ağırlık dosyalarını bir input vererek SAHI ile test etmek istiyorum.

**Reply by fcakyon** [reply]
> **Düzeltme İsteği:** Bu pathi düzeltebilir misin? https://github.com/fcakyon/yolov5-pip/blob/main/yolov5/helpers.py#L333

O kod blogunun hic orada olmamasi gerekiyor, tesekkurler uyari icin, kaldiriyorum: https://github.com/fcakyon/yolov5-pip/pull/183

**Reply by fcakyon** [reply]
> **Soru:** Helpers.py dosyasındaki train_utils fonksiyonları HF platformunda train yapabilmek için mi?

train_utils fonksiyonları derken hangi fonksiyonlari kastediyorsun? HF platformunda egitim yapabilmek bir ekleme yapmadim.

**Reply by fcakyon** [reply]
> **Özellik İsteği:** https://huggingface.co/spaces/fcakyon/sahi-yolov5/ reposuna bu desteklere ekleyebilir misiniz? HF platformundaki ağırlık dosyalarını bir input vererek SAHI ile test etmek istiyorum.

Su anda SAHI bu ozelligi destekliyor mu emin degilim, denemek/incelemek lazim.

**Reply by kadirnar** [reply]
> > **Soru:** Helpers.py dosyasındaki train_utils fonksiyonları HF platformunda train yapabilmek için mi?
> 
> train_utils fonksiyonları derken hangi fonksiyonlari kastediyorsun? HF platformunda egitim yapabilmek bir ekleme yapmadim.

https://github.com/fcakyon/yolov5-pip/blob/main/yolov5/helpers.py#L86-L329

**Reply by fcakyon** [reply]
> > > **Soru:** Helpers.py dosyasındaki train_utils fonksiyonları HF platformunda train yapabilmek için mi?
> > 
> > 
> > train_utils fonksiyonları derken hangi fonksiyonlari kastediyorsun? HF platformunda egitim yapabilmek bir ekleme yapmadim.
> 
> https://github.com/fcakyon/yolov5-pip/blob/main/yolov5/helpers.py#L86-L329

`convert_coco_dataset_to_yolo` ve `upload_to_s3` fonksiyonlari onceden train.py icinde vardi buraya tasidim sadece.

digerleri huggingface hub'a yolov5 training sonuclarini otomatik olarak yuklemek icin.

**Reply by kadirnar** [reply]
Bilgilendirme için teşekkür ederim.
